### PR TITLE
Add support for checking type of primitives for creating a bounding volume hierarchy

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: packages/Search/src/details/DTK_DetailsConcepts.hpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    Geometry, Other,
+    std::void_t<decltype(
+        expand( std::declval<Geometry &>(), std::declval<Other const &>() ) )>>
+    : std::true_type
+{
+};
+
+// Checks for existence of a free function that calculates the centroid of an
+// object of type Geometry
+template <typename Geometry, typename Point, typename = void>
```
Comment (Rombur) [inline packages/Search/src/details/DTK_DetailsConcepts.hpp]: Why is `Point` a template?

Comment (dalg24) [reply packages/Search/src/details/DTK_DetailsConcepts.hpp]: Might be overkill at the moment.  I had in mind to allow customization.  We'd only be able to access the coordinates through some traits when we compute Morton codes, we don't actually need the type to be our `DTK::Point`.  Also didn't feel like including any other DTK headers.
You want me to change it?

**Comment by dalg24** [comment]
>  can you squash a few commits (fix typo, remove type modification,...)

Okay, will clean history just before the merge.

File: packages/Search/src/details/DTK_DetailsConcepts.hpp (hunk 1 above)
Comment (Rombur) [reply packages/Search/src/details/DTK_DetailsConcepts.hpp]: No, that's fine. I don't have a strong opinion on this.

**Comment by dalg24** [comment]
(Rebased on #507 and pushed force)

**Pull request merged**